src/iOS/Xamarin.iOS/Samples/Layers/QueryCQLFilters/QueryCQLFilters.cs

```diff
@@ -0,0 +1,387 @@
+        }
+
+        private void ClauseSelected(UIAlertAction obj)
+        {
+            _whereClause = obj.Title;
+            _clauseButton.SetTitle(obj.Title, UIControlState.Normal);
+        }
+
+        public override void LoadView()
+        {
+            // Create the views.
+            View = new UIView { BackgroundColor = ApplicationTheme.BackgroundColor };
+
+            _myMapView = new MapView();
+            _myMapView.TranslatesAutoresizingMaskIntoConstraints = false;
+
+            _activityIndicator = new UIActivityIndicatorView(UIActivityIndicatorViewStyle.WhiteLarge)
+            {
+                TranslatesAutoresizingMaskIntoConstraints = false,
+                HidesWhenStopped = true,
+                BackgroundColor = UIColor.FromWhiteAlpha(0, .5f)
+            };
+
+            UILabel clauseLabel = new UILabel { Text = "Where clause:", TranslatesAutoresizingMaskIntoConstraints = false };
+            _clauseButton = new UIButton() { TranslatesAutoresizingMaskIntoConstraints = false };
+            _clauseButton.SetTitle("Select Clause", UIControlState.Normal);
+            _clauseButton.SetTitleColor(View.TintColor, UIControlState.Normal);
+            var clauseStack = GetRowStackView(new UIView[] { clauseLabel, _clauseButton });
+
+            UILabel maxLabel = new UILabel { Text = "Maximum features:", TranslatesAutoresizingMaskIntoConstraints = false };
+            _maxFeaturesEntry = new UITextField { TranslatesAutoresizingMaskIntoConstraints = false, Text = "3000" };
+            var maxFeaturesStack = GetRowStackView(new UIView[] { maxLabel, _maxFeaturesEntry });
+
+            UILabel dateLabel = new UILabel { Text = "Time extent:", TranslatesAutoresizingMaskIntoConstraints = false };
+            _dateSwitch = new UISwitch { TranslatesAutoresizingMaskIntoConstraints = false, On = true };
+            var dateSwitchStack = GetRowStackView(new UIView[] { dateLabel, _dateSwitch });
+
+            _startDatePicker = new UIDatePicker { PreferredDatePickerStyle = UIDatePickerStyle.Compact, Mode = UIDatePickerMode.Date };
+            _endDatePicker = new UIDatePicker { PreferredDatePickerStyle = UIDatePickerStyle.Compact, Mode = UIDatePickerMode.Date };
+            var datesStack = GetRowStackView(new UIView[] { _startDatePicker, _endDatePicker });
+
+            _applyButton = new UIBarButtonItem { Title = "Apply" };
+            _resetButton = new UIBarButtonItem { Title = "Reset" };
+
+            _applyToolbar = new UIToolbar { TranslatesAutoresizingMaskIntoConstraints = false };
+            _applyToolbar.Items = new[]
+            {
+                _applyButton,
+                new UIBarButtonItem(UIBarButtonSystemItem.FlexibleSpace),
+                _resetButton
+            };
+
+            // Add the views.
+            View.AddSubviews(_myMapView, _activityIndicator, dateSwitchStack, maxFeaturesStack, clauseStack, datesStack, _applyToolbar);
```

Comment:
Activity indicator should be added last to ensure that it is on top of the other views; currently it is possible to interact with the sample while things are loading.